test/test_schedule.py

```diff
@@ -169,15 +169,15 @@ def test_fold_conv_relu(self):
     # run
     img = Tensor.ones(2,3,64,64)
     out = c1(img).relu()
-    check_schedule(out, 1, [c1.weight, c1.bias])
+    check_schedule(out, 3, [c1.weight, c1.bias])
```

Comment:
Why is this increasing?

Reply:
this was increasing because BinaryOp was rerunning after reshape was pushed up. I had restricted this behaviour to an extend via "not all(x.realized or len(x.op.src) == 0 for x in self.op.buffers)" but this wasn't sufficient. 

Using "map(functools.partial(get_movementroot, allow_contiguous = True), self.op.buffers))"  instead now!

Reply:
attaching for reference, the re-rerunning of BinOp.
![image](https://github.com/tinygrad/tinygrad/assets/88420255/f0860480-37c7-48a3-b424-03376c1fffab)